[DebugInfo] Set the autogenerated bit to request recycled locations

vedantk opened the pull request:
IRGenDebugInfo expects to see an autogenerated bit before allowing reuse
of the last seen debug location for a new instruction.

rdar://28544248

vedantk committed:
[DebugInfo] Remove a default location argument from setCurrentLoc

Having a default location of None doesn't seem to enhance readability.

vedantk committed:
[DebugInfo] Set the autogenerated bit to request recycled locations

IRGenDebugInfo expects to see an autogenerated bit before allowing reuse
of the last seen debug location for a new instruction.

rdar://28544248

vedantk commented:
@swift-ci Please smoke test

adrian-prantl commented on lib/IRGen/IRGenDebugInfo.cpp:
```diff
@@ -125,7 +125,7 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
 
   void setCurrentLoc(IRBuilder &Builder, const SILDebugScope *DS,
-                     Optional<SILLocation> Loc = None);
+                     Optional<SILLocation> Loc);
```
Is there a caller left that actually passes None? Otherwise we could drop the Optional.

vedantk replied:
Good call, I'll do that.

adrian-prantl reviewed (approved):
Looks good!

vedantk committed:
[DebugInfo] Remove option to pass None into setCurrentLoc, NFC

Passing in some SILLocation to IRGenDebugInfo setCurrentLoc() will now
be mandatory.

vedantk commented:
@swift-ci Please smoke test and merge

vedantk commented:
@swift-ci Please smoke test and merge
